ufo2fontir/src/source.rs

```diff
@@ -478,35 +474,66 @@ fn files_identical(f1: &Path, f2: &Path) -> Result<bool, WorkError> {
     Ok(true)
 }
 
-fn font_infos(
+/// Creates a map from UFO directory name => fontinfo.
+///
+/// That is, source.filename => fontinfo.
+fn font_infos<'a>(
     designspace_dir: &Path,
-    designspace: &DesignSpaceDocument,
-) -> Result<Vec<plist::Dictionary>, WorkError> {
-    let mut results = Vec::new();
+    designspace: &'a DesignSpaceDocument,
+) -> Result<HashMap<&'a String, norad::FontInfo>, WorkError> {
+    let mut results = HashMap::new();
     for source in designspace.sources.iter() {
         let ufo_dir = designspace_dir.join(&source.filename);
-        if !ufo_dir.join("fontinfo.plist").exists() {
-            continue;
-        }
-        results.push(load_plist(&ufo_dir, "fontinfo.plist")?);
+        let data_request = norad::DataRequest::none();
+        let font = norad::Font::load_requested_data(&ufo_dir, data_request)
+            .map_err(|e| WorkError::ParseError(ufo_dir, format!("{e}")))?;
+        results.insert(&source.filename, font.font_info);
     }
     Ok(results)
 }
 
+fn names(font_info: &norad::FontInfo) -> HashMap<NameKey, String> {
+    let mut result = HashMap::new();
+
+    let mut insert_if_present = |maybe_value: &Option<String>, name_id: NameId| {
+        if let Some(value) = maybe_value.as_ref() {
+            // For now let's very optimistically emit unicode names only
+            // <https://learn.microsoft.com/en-us/typography/opentype/spec/name#platform-encoding-and-language-ids>
+            result.insert(
+                NameKey {
+                    platform_id: 0, // Unicode
```

Comment:
note that fontmake (ufo2ft) emits "Windows" platform (3) name records only, which also support the full unicode repertoire, and AFAIK they should be more widely supported than strict "Unicode" platform name records.

> When building a Unicode font for Windows, the platform ID should be 3 and the encoding ID should be 1... All string data for platform 3 must be encoded in UTF-16BE.

https://learn.microsoft.com/en-us/typography/opentype/spec/name#windows-encoding-ids

ufo2ft defaults to platform_id=3, encoding_id=1 if name string is BMP-only else 10, langID=0x409 (for American English):
https://github.com/googlefonts/ufo2ft/blob/fca66fe3ea1ea88ffb36f8264b21ce042d3afd05/Lib/ufo2ft/outlineCompiler.py#L430-L432

We should match that.

re: english only, note that multilingual name strings can also be set using `openTypeNameRecords` entries in fontinfo.plist. You can add support for those later on if you like.

Reply:
I'm not spotting how to access l10n values of a name from https://docs.rs/norad/latest/norad/fontinfo/struct.FontInfo.html. @cmyr ?

Reply:
Oh nevermind, apparently UFO has individual fields that don't specify platform, encoding, lang *and* a list of names with platform, encoding, lang. I was thrown by the individual field with no apparent way to set those fields.